### PR TITLE
Change default hydrogen mass

### LINKED ISSUE #982: Update default hydrogen mass to 3 amu for protein mutations
Using 4 amu causes stochastic nans for some transformations, e.g. ALA->TYR where hydrogen mass repartitioning with 4 amu causes the CB to have a mass of 4.53 amu and each of the HBs have a mass of 4 amu, which leads to instabilities, particularly when running with rest over the protocol.
Note that this has already been done for small molecules.

### PR DESCRIPTION
**Pull request by zhang-ivy**
## Description

Change default hydrogen mass in all files (including tests). Mostly doing this for protein mutations, but also necessary for small molecule tests, though this was already changed for in the small molecule setup pipeline earlier.

## Motivation and context

Resolves #982 

## Change log

```

```

### CODE REVIEW

**Review by jchodera — APPROVED** [review]
Thanks! This should improve stability!

File: perses/app/relative_point_mutation_setup.py [hunk 1]
```diff
@@ -52,7 +52,7 @@ class PointMutationExecutor(object):
                                      ligand_index=0,
                                      forcefield_files=['amber14/protein.ff14SB.xml', 'amber14/tip3p.xml'],
                                      barostat=openmm.MonteCarloBarostat(1.0 * unit.atmosphere, temperature, 50),
-                                     forcefield_kwargs={'removeCMMotion': False, 'ewaldErrorTolerance': 1e-4, 'nonbondedMethod': app.PME, 'constraints' : app.HBonds, 'hydrogenMass' : 4 * unit.amus},
+                                     forcefield_kwargs={'removeCMMotion': False, 'ewaldErrorTolerance': 1e-4, 'nonbondedMethod': app.PME, 'constraints' : app.HBonds, 'hydrogenMass' : 3 * unit.amus},
```
Comment (jchodera) [inline perses/app/relative_point_mutation_setup.py:55]: @zhang-ivy : Do `forcefield_kwargs` get dispatched to `openmm.app.ForceField` or to `openmmtools.generators.SystemGenerator`? If `SystemGenerator`, then `ewaldErrorTolerance` and `nonbondedMethod` should be in `periodic_forcefield_kwargs`. Worth double-checking to make sure we don't have issues with this elsewhere.

**Review by ijpulidos — CHANGES_REQUESTED** [review]
Looks good! I could only spot one missing in https://github.com/choderalab/perses/blob/main/perses/app/relative_setup.py#L57

File: perses/app/relative_point_mutation_setup.py (hunk 1 above)
Comment (zhang-ivy) [reply perses/app/relative_point_mutation_setup.py:57]: They are passed into `SystemGenerator`. This docstring is not up to date. `nonbondedMethod` has already been moved to `periodic_forcefield_kwargs`, but I'll need to move `ewaldErrorTolerance` as well. Thanks!

**Comment by zhang-ivy** [comment]
@ijpulidos : The code you linked above already has the hydrogen mass = 3 amu. Did you mean to link a different line? 

If there aren't any lines I forgot to change to 3 amu, could you review one more time and merge? Thanks!

**Comment by jchodera** [comment]
@ijpulidos : Can you review this and merge when able?

**Review by ijpulidos — APPROVED** [review]
Looks good to me. LGTM!

**Pull request merged**